pandas/tests/indexes/interval/test_base.py

```diff
@@ -79,3 +79,10 @@ def test_where(self, closed, klass):
         expected = IntervalIndex([np.nan] + idx[1:].tolist())
         result = idx.where(klass(cond))
         tm.assert_index_equal(result, expected)
+
+    def test_getitem_2d_deprecated(self):
+        # GH#30588 multi-dim indexing is deprecated, but raising is also acceptable
+        idx = self.create_index()
+        with pytest.raises(ValueError, match="multi-dimensional indexing not allowed"):
+            with tm.assert_produces_warning(DeprecationWarning, check_stacklevel=False):
+                idx[:, None]
```

Comment:
Shouldn't this only raise the error? (while trying out this branch, for me it also does)

Reply:
Keep in mind, it's a deprecation warning and may not be visible.

Reply:
Yep, that's it

```python

In [11]: warnings.simplefilter("always", DeprecationWarning)

In [12]: idx[np.array([[0, 1]])]
/Users/taugspurger/sandbox/pandas/pandas/core/arrays/interval.py:498: DeprecationWarning: Support for multi-dimensional indexing (e.g. `index[:, None]`) on an Index is deprecated and will be removed in a future version.  Convert to a numpy array before indexing instead.
  left = self.left[value]
/Users/taugspurger/sandbox/pandas/pandas/core/arrays/interval.py:499: DeprecationWarning: Support for multi-dimensional indexing (e.g. `index[:, None]`) on an Index is deprecated and will be removed in a future version.  Convert to a numpy array before indexing instead.
  right = self.right[value]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-bd6337c200ee> in <module>
----> 1 idx[np.array([[0, 1]])]

~/sandbox/pandas/pandas/core/indexes/interval.py in __getitem__(self, value)
    990
    991     def __getitem__(self, value):
--> 992         result = self._data[value]
    993         if isinstance(result, IntervalArray):
    994             return self._shallow_copy(result)

~/sandbox/pandas/pandas/core/arrays/interval.py in __getitem__(self, value)
    505             if np.ndim(left) > 1:
    506                 # GH#30588 multi-dimensional indexer disallowed
--> 507                 raise ValueError("multi-dimensional indexing not allowed")
    508             return Interval(left, right, self.closed)
    509

ValueError: multi-dimensional indexing not allowed

```

I think that's OK for now, but can fix up later if desired.